devices/debian_wifi.py

```diff
@@ -7,29 +7,38 @@ class DebianWifi(debian.DebianBox):
 
     model = ('debianwifi')
 
-    iface_wlan = "wlan1"
-
-    def scan(self):
+    def wifi_scan(self, iface_wlan="wlan1", login_pwd=None):
         from tests.lib.installers import install_iw
         install_iw(self)
 
-        self.sendline('iw %s scan | grep SSID:' % self.iface_wlan)
+        self.sendline('iw %s scan | grep SSID:' % iface_wlan)
         self.expect(self.prompt)
-
+        match = re.search("Operation not permitted", self.before)
+        if match:
+            self.sendline('sudo iw %s scan | grep SSID:' % iface_wlan)
+            idx = self.expect(["password for .*:"] + self.prompt,timeout=20)
+            if idx == 0:
+                self.sendline(login_pwd)
+                self.expect(self.prompt)
         return self.before
 
-    def wpa_connect(self, ssid_name, password):
+    def wpa_connect(self, ssid_name, password, iface_wlan="wlan1", login_pwd=None):
          '''Generate WPA supplicant file and execute it'''
-         self.sendline("rm /etc/"+ssid_name+".conf")
+         self.sendline("rm "+ssid_name+".conf")
+         self.expect(self.prompt)
+         self.sendline("wpa_passphrase "+ssid_name+" "+password+" >> "+ssid_name+".conf")
          self.expect(self.prompt)
-         self.sendline("wpa_passphrase "+ssid_name+ " >> /etc/"+ssid_name+".conf")
-         self.expect("")
-         self.sendline(password)
-         self.expect(self.prompt,timeout=20)
-         self.sendline("cat /etc/"+ssid_name+".conf")
+         self.sendline("cat "+ssid_name+".conf")
          self.expect(self.prompt)
-         self.sendline("wpa_supplicant  -B -Dnl80211 -i"+self.iface_wlan+ " -c/etc/"+ssid_name+".conf")
+         self.sendline("wpa_supplicant  -B -Dnl80211 -i"+iface_wlan+ " -c"+ssid_name+".conf")
```

Comment:
Yea, so why are we switching away? If there is a new one, we need to handle two radios somehow...